rpc/core/tx.go

```diff
@@ -19,36 +19,44 @@ import (
 // More: https://docs.tendermint.com/master/rpc/#/Info/tx
 func (env *Environment) Tx(ctx *rpctypes.Context, hash []byte, prove bool) (*ctypes.ResultTx, error) {
 	// if index is disabled, return error
-	if _, ok := env.TxIndexer.(*null.TxIndex); ok {
-		return nil, fmt.Errorf("transaction indexing is disabled")
-	}
 
-	r, err := env.TxIndexer.Get(hash)
-	if err != nil {
-		return nil, err
+	if !indexer.KVSinkEnabled(env.EventSinks) {
+		return nil, errors.New("transaction querying is disabled due to no kvEventSink")
 	}
 
-	if r == nil {
-		return nil, fmt.Errorf("tx (%X) not found", hash)
-	}
+	for _, sink := range env.EventSinks {
+		if sink.Type() == indexer.KV {
+			r, err := sink.GetTxByHash(hash)
+			if err != nil {
+				return nil, err
+			}
+
+			if r == nil {
+				return nil, fmt.Errorf("tx (%X) not found", hash)
+			}
+
+			height := r.Height
+			index := r.Index
 
-	height := r.Height
-	index := r.Index
+			var proof types.TxProof
+			if prove {
+				block := env.BlockStore.LoadBlock(height)
+				proof = block.Data.Txs.Proof(int(index)) // XXX: overflow on 32-bit machines
+			}
 
-	var proof types.TxProof
-	if prove {
-		block := env.BlockStore.LoadBlock(height)
-		proof = block.Data.Txs.Proof(int(index)) // XXX: overflow on 32-bit machines
+			return &ctypes.ResultTx{
+				Hash:     hash,
+				Height:   height,
+				Index:    index,
+				TxResult: r.Result,
+				Tx:       r.Tx,
+				Proof:    proof,
+			}, nil
+		}
 	}
 
-	return &ctypes.ResultTx{
-		Hash:     hash,
-		Height:   height,
-		Index:    index,
-		TxResult: r.Result,
-		Tx:       r.Tx,
-		Proof:    proof,
-	}, nil
+	return nil, errors.New("transaction querying is disabled on this node due to the KV event sink being disabled;" +
+		" please see config.toml if this is unexpected")
```

Comment:
nit: I'd use `fmt.Sprint` over string contamination because it uses a buffer pool

also maybe just drop the "please see" part.